src/parser.rs

```diff
@@ -988,16 +988,25 @@ impl Parser {
     /// Parse a SQL CREATE statement
     pub fn parse_create(&mut self) -> Result<Statement, ParserError> {
         if self.parse_keyword(Keyword::TABLE) {
-            self.parse_create_table()
+            self.parse_create_table(false)
+        } else if self.parse_keywords(&[Keyword::OR, Keyword::REPLACE, Keyword::TABLE]) {
+            self.parse_create_table(true)
```

Comment:
Would this be clearer as:

```
let or_replace = self.parse_keywords(&[Keyword::OR, Keyword::REPLACE]);
if self.parse_keyword(Keyword::TABLE) {
   self.parse_create_table(or_replace)
} ...
```
?

Reply:
Thought about it too, however tries to parse "or replace" for every other alternative too.
Code is a bit cleaner though.

Reply:
If by "tries to parse" you mean "accepts", that doesn't have to be the case. I meant 
```
let or_replace = self.parse_keywords(&[Keyword::OR, Keyword::REPLACE]);
if self.parse_keyword(Keyword::TABLE) {
   self.parse_create_table(or_replace)
} else if self.parse_keyword(Keyword::VIEW) {
   self.parse_create_table(or_replace)
} else if or_replace {
  // error
} else if self.parse_keyword(Keyword::INDEX) {
  ...
```

If you literally mean *tries*, then this https://github.com/ballista-compute/sqlparser-rs/pull/239/files#diff-2c09afcdc3c420ab0678ba9b5e83959cR992 already tries to parse "OR REPLACE" for every alternative.

Reply:
Your current version silently eats "OR REPLACE" with some object types, like CREATE INDEX - is that intentional?

Reply:
Hm that's right, after changing it with the 

`let or_replace = self.parse_keywords(&[Keyword::OR, Keyword::REPLACE]);`

it will consume those. Maybe better to revert to the first implementation, or do you have some other suggestion?

Reply:
Ah I see the error part, will do that.

Reply:
Resolved now